### DISCUSSION TITLE
Anyway to show progress of search ?

### DISCUSSION THREAD

**Discussion by iksi4prs** [sharkdp/fd#1514] [Q&A] [answered]
Hi,
Sometimes I search for a file in a very large collection,
and the 1st result maybe be found only after few minutes.
I know the app is still running, and not stuck, but it would be nice to have some visual indication.

Is it possible to show also progress of searched files , (not only search results , for that I can use the "--exec" flag) ?

Eg:
`23/4444 ******`
meaning: 4444 searched, 23 results, and the stars are some minimal progress animation.

Thanks for your help,
Si.

**Comment by tavianator** [comment, marked as answer]
`fd` doesn't have a progress indicator.  See #740 for some prior discussion.

[`bfs`](https://github.com/tavianator/bfs) has one (`-status`).  But it uses `find` syntax, not `fd`'s syntax.

**Reply by iksi4prs** [reply]
@tavianator ,
Thanks, I see #740 was closed. I guess because of option to use -x to show progress of results, but I'm interested in progress of searches.
Thanks for tip about bfs, I will check it.

**Reply by iksi4prs** [reply]
@tavianator,
I have one more question, in case I want to try and implement it.
I've checked the code, and the number of results is easy to find in in walk.rs `self.num_results`.
Am I correct to assume the big task for implementing this, is to collect the number of 'searched' files (in code `return WalkState::Continue`) ?
Because now I see there are only `WorkerResult::Entry `and `WorkerResult::Error` reported back.

**Reply by tavianator** [reply]
Yeah that'd be the idea.  But it's probably too slow to send something over the channel for every file.  Better to increment an atomic counter.  Ideally a separate counter for each thread.
